katcp/resource_client.py

```diff
@@ -55,6 +55,23 @@ def _transform(f):
     future.add_done_callback(_transform)
     return new_future
 
+def until_all(futures):
+    """Run multiple operations in parallel but only raise a single TimeoutError.
+
+    This returns a yieldable object that will contain the results of `futures`
+    in an equivalent container (list or dict). However, it only raises the
+    first TimeoutError and ignores any further timeouts. This makes it useful
+    to wait on the results of multiple operations with the same timeout, as the
+    first occurrence of a timeout means that any unresolved futures will also
+    necessarily time out, leading to a rash of duplicate exceptions otherwise.
```

Comment:
> as the first occurrence of a timeout means that any unresolved futures will also necessarily time out

Is bit of an assumption ... but ok I guess, as long as the other future calls will get a chance to finish and not be interrupted by the first future call with a TimeoutError - its not clear to me from the documentation if this is the case though:
> If any children raise exceptions, `multi(`) will raise the first one. All others will be logged, unless they are of types contained in the `quiet_exceptions` argument.